In [2]:
# Normal Distribution
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt 
from scipy.stats import chi2

mean = 85
ST_D = 10 
pop = 100
n = 10 #Sqrt(n)

exam = np.random.normal(mean, ST_D, n).round()
exam
mean_exam = exam.mean()
mean_exam # sample interval?

86.5

In [3]:
# Confidence Intervals

SE = ST_D / (mean_exam**.5) #SQRT of sample pop
print (SE) # Hand calculated, not sure if we where to use this instead: standard_error = stats.sem(exam_mean)

1.0752066611409408


In [4]:
# Upper and lower limits

standard_deviation = 1.96
upper_limit = mean_exam + (standard_deviation * SE) 
lower_limit = mean_exam - (standard_deviation * SE)
print('Upper:', upper_limit)
print('Lower:', lower_limit)

Upper: 88.60740505583624
Lower: 84.39259494416376


From this we can say that the students in the class will score will be slightly lower than the mean and are about 5 points away from each other. I'm not sure about my SE calculation but it seems like the class wil under perform the mean of 85. Based on our interval/sample (exam_mean) we can conclude the populations' mean is between our upper and lower intervals

In [5]:
# Chi-Squared

meta = np.array([ [ 10, 4, 10 ], [ 0, 6, 6 ] ])
stats.chi2_contingency(meta)

(8.325000000000001,
 0.015568587758182073,
 2,
 array([[ 6.66666667,  6.66666667, 10.66666667],
        [ 3.33333333,  3.33333333,  5.33333333]]))

First line is our Chi Squared value: Difference between expected and observed values: (8.325)

Second is our p-value: (0.0156) Significant**

Third is our degrees of freedom or sample size: (2)

Lastly is an expected/probable distribution: These are ranges for the scores based on the Authors who wrote them

From this we can tell that the P-Value is significant so we reject the null hypothesis and the two observed are similar as expected

In [6]:
# Anova 

means = 100
sizes = 500
standard_dev = [10, 50]


test1 = np.random.normal(means,  standard_dev[0], sizes).round()
test2 = np.random.normal(means,  standard_dev[1], sizes).round()
stats.f_oneway(test1, test2)

F_onewayResult(statistic=1.8591429056988171, pvalue=0.1730312394560325)

Our first value is our F-Value, variation between the sample means or variation within samples: (.5867) 

Second is our P-Value ( 0.4439 > 0.05 ) 

so we reject the null hypothesis and say that the testing groups are similar to each other